src/Microsoft.Graph.Core/Requests/Middleware/RetryHandler.cs

```diff
@@ -129,34 +128,47 @@ private void AddOrUpdateRetryAttempt(HttpRequestMessage request, int retry_count
         /// </summary>
         /// <param name="response">The <see cref="HttpResponseMessage"/>returned.</param>
         /// <param name="retry_count">The retry counts</param>
+        /// <param name="delay">Delay value in seconds.</param>
         /// <param name="cancellationToken">The cancellationToken for the Http request</param>
         /// <returns>The <see cref="Task"/> for delay operation.</returns>
-        public Task Delay(HttpResponseMessage response, int retry_count, CancellationToken cancellationToken)
+        public Task Delay(HttpResponseMessage response, int retry_count, int delay, CancellationToken cancellationToken)
         {
-            
-            TimeSpan delay = TimeSpan.FromMilliseconds(0);
+            //TimeSpan delayTimeSpan = TimeSpan.FromMilliseconds(0);
             HttpHeaders headers = response.Headers;
+            double delayInSeconds = RetryOption.Delay;
             if (headers.TryGetValues(RETRY_AFTER, out IEnumerable<string> values))
             {
                 string retry_after = values.First();    
                 if (Int32.TryParse(retry_after, out int delay_seconds))
                 {
-                    delay = TimeSpan.FromSeconds(delay_seconds);
+                    delayInSeconds = delay_seconds;
                 }
             }
             else
             {
+                m_pow = Math.Pow(2, retry_count);
+                delayInSeconds = m_pow * RetryOption.Delay;
+            }
 
-                m_pow = Math.Pow(2, retry_count); // m_pow = Pow(2, retry_count)
+            TimeSpan delayTimeSpan = TimeSpan.FromSeconds(Math.Min(delayInSeconds, RetryHandlerOption.MAX_DELAY));
 
-                double delay_time = m_pow * DELAY_MILLISECONDS;
-              
-                delay = TimeSpan.FromMilliseconds(delay_time);
-            }
-            return Task.Delay(delay, cancellationToken);
+            return Task.Delay(delayTimeSpan, cancellationToken);
 
         }
 
-
+        /// <summary>
+        /// Check the HTTP response's status to determine whether it should be retried or not.
+        /// </summary>
+        /// <param name="response">The <see cref="HttpResponseMessage"/>returned.</param>
+        /// <returns></returns>
+        private bool IsRetry(HttpResponseMessage response)
```

Comment:
This should be named ShouldRetry. IsRetry give me the impression that the response comes from a retry, not that we should retry based on the results in the response.